p2panda-rs/src/operation/verified_operation.rs

```diff
@@ -27,12 +27,14 @@ pub struct VerifiedOperation {
     operation: Operation,
 }
 
-impl VerifiedOperation {
+impl AsVerifiedOperation for VerifiedOperation {
+    type VerifiedOperationError = VerifiedOperationError;
+
     /// Returns a new `VerifiedOperation` instance.
     ///
     /// Use `VerifiedOperation::new_from_entry()` instead if you want to validate that the operation
     /// was signed by this public key.
-    pub fn new(
+    fn new(
```

Comment:
I'm still a bit unsure about documenting traits vs documenting structs but maybe trait items like this should have a description of what implementations need to verify - or don't need to verify. In this case it could say something like:

> Implementations should document that this method doesn't validate whether it's parameters match each other and that consumers must ensure that parameters were validated before using this constructor.

Reply:
Yes, I like that!

Reply:
I assumed you meant changing the docs in the trait definition?

Reply:
Hmm, also, thinking again. Not validating here isn't _compulsory_, if an implementation can do some validation somehow, this is fine. The thing we want to strongly encourage is that `new_from_entry)` implementations _do_ perform validation.

Reply:
What do you think @cafca ? Is this good to go as is?

Reply:
Yes!